api/test/test_play_asset_pack_manager.gd

```diff
@@ -161,3 +162,117 @@ func test_cancel_asset_pack_request_non_existing_pack_name():
 	
 	var test_result : bool = test_object.cancel_asset_pack_request(test_pack_name)
 	assert_eq(test_result, false)
+
+func test_show_cellular_data_confirmation_success():
+	var mock_plugin = FakeAndroidPlugin.new()
+
+	# configure what should be emitted upon show_cellular_data_confirmation() call
+	mock_plugin.set_show_confirmation_response(true, PlayAssetPackManager.AssetPackStorageMethod.STORAGE_FILES, {})
+	var test_object = create_play_asset_pack_manager(mock_plugin)
+	
+	var request_object = test_object.show_cellular_data_confirmation()
+	
+	# connect to helper function, simulating the workflow of connecting callback to signal
+	request_object.connect("request_completed", self, "show_cellular_data_confirmation_success_helper")
+	
+	# yield to the request_completed signal for no longer than 1 seconds and assert for signal emitted
+	yield(yield_to(request_object, "request_completed", 1), YIELD)
+	assert_signal_emitted(request_object, "request_completed", "signal should have emitted")
+	
+	# assert using getters, simulating the workflow of yielding the signals
+	assert_true(request_object.get_status())
```

Comment:
```show_cellular_data_confirmation_success_helper()``` will be called automatically upon ```emit_signal``` since it is already connected in line 176. The assertions below is also needed since we need to test the workflow for yielding to signals.

Reply:
The ```	yield(yield_to(request_object, "request_completed", 1), YIELD) ````statement also asserts the ```request_completed``` must be received and the method connected must be called.